server/http/server.go

```diff
@@ -0,0 +1,74 @@
+package httpserver
```

Comment:
this is in `server/` and not `pkg/`, shouldn't it be?

Reply:
Ah, you know what? While making this PR I had the thought that the `pkg` directory was only for the things we wanted to export in the library. I'll move both the `server` and `cli` directories into `pkg`.

Reply:
I actually agree with Kyle's original decision and would prefer we revert it. 

Anything that isn't "library code" -- i.e. the thing you'd use on your own if you already had an executable, should not be in package.

I'd like to run through some scenario use cases:

- I am an end user who wants to retrieve with a CLI
- I am a different language user who wants to use this library through HTTP
- I have a go application using none of the Libp2p/IPFS/Filecoin stack and I want to incorporate IPFS/Filecoin retrievals through this library
- I have an existing Libp2p application which uses Bitswap and/or GraphSync and I want to incorporate this library for doing retrievals

My feeling is that only in the last case am I using only code in `/pkg`. My feeling is that `/server`, `/cli`, and even `/lassie` (I might rename this something like `node` to capture that it's a complete stack of retrieval code) should be in the top level dir.

Reply:
`/server` was moved back outside of `/pkg` and `/cli` was removed in favor of putting `flags.go` directory into `/cmd/lassie`